misc: reorganize the configuration file of nydusd

adam3q opened the pull request:
Fix: #1180
1. Move configuration files from docs/samples to misc/configs
2. Fix incomplete configuration in docs/nydusd.md
3. Update outdated nydusd-config.json from nydus-snapshotter repo

adamqqqplay commented on docs/nydusd.md:
```diff
@@ -8,27 +8,42 @@ Get `nydusd` binary from [release](https://github.com/dragonflyoss/image-service
+  "iostats_files": false,
+  "enable_xattr": true,
+  "fs_prefetch": {
+    "enable": true,
```
@bergwolf I copied the config file from [here](https://github.com/containerd/nydus-snapshotter/blob/main/misc/snapshotter/nydusd-config-localfs.json), are these options unnecessary?

adamqqqplay replied:
@bergwolf I removed this part from config.

jiangliu commented on docs/nydusd.md:
```diff
@@ -8,27 +8,41 @@ Get `nydusd` binary from [release](https://github.com/dragonflyoss/image-service
       "type": "localfs",
       "config": {
-        "dir": "/path/to/blobs",
+        "dir": "/tmp/blobs/"
```
how about using `/var/lib/nydus/blobs` here?

adamqqqplay replied:
Looks good. I fixed it.

jiangliu commented on misc/configs/nydusd-config.json:
```diff
@@ -3,25 +3,23 @@
     },
     "cache": {
-      "type": "blobcache",
-      "config": {
```
why the `config` field get removed?

adamqqqplay replied:
@jiangliu After this [commit](https://github.com/containerd/nydus-snapshotter/commit/c5e1c57766aa4ac96fe80f7ec2c09223af72d845) in https://github.com/containerd/nydus-snapshotter/pull/229, the `work_dir` field is unnecessary.

adamqqqplay replied:
@jiangliu I misunderstood before, this field is still required when there is no snapshotter, it has been fixed now.

adam3q committed:
misc: reorganize the configuration file of nydusd

1. Move configuration files from docs/samples to misc/configs
2. Fix incomplete configuration in docs/nydusd.md
3. Update outdated nydusd-config.json from nydus-snapshotter repo

Signed-off-by: Qinqi Qu <quqinqi@linux.alibaba.com>
